test/parallel/test-fs-promises-file-handle-read-worker.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+const common = require('../common');
+const fs = require('fs');
+const assert = require('assert');
+const path = require('path');
+const tmpdir = require('../common/tmpdir');
+const file = path.join(tmpdir.path, 'read_stream_filehandle_worker.txt');
+const input = 'hello world';
+const { Worker, isMainThread, workerData } = require('worker_threads');
+
+let output = '';
+
+if (isMainThread) {
```

Comment:
```suggestion
if (isMainThread || workerData === undefined) {
```

Reply:
Yeah, but why? Do you understand what is happening? I built a Node-master on Ubuntu 16.04 and I do not reproduce it.
I also went through the other tests and I saw that some of them had that check, but others did not and still passed.

Reply:
@mmomtchev https://github.com/nodejs/node/pull/35922#issuecomment-734382138

If you want to reproduce locally, use `python tools/test.py --worker parallel/test-fs-promises-file-handle-read-worker`

Reply:
@aduh95  @addaleax in fact, `--worker` simply launches all the tests in a worker context :smile: 
So it is absolutely normal for my code to behave in a very strange way - `isMainThread` is `false` from the start and the control goes directly into the worker thread
thanks for the tip